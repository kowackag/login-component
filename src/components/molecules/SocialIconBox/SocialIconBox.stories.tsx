import type { Meta, StoryObj } from "@storybook/react";

import { SocialIconBox } from "./SocialIconBox";

const meta: Meta<typeof SocialIconBox> = {
	component: SocialIconBox,
};

export default meta;

type Story = StoryObj<typeof SocialIconBox>;

export const Default: Story = {
	args: {},
};
