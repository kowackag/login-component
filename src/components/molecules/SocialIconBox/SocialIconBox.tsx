import { IconLink } from "@/components/atoms/IconLink/IconLink";

export const SocialIconBox = () => {
	return (
		<div data-testid="icon-box" className="flex justify-center">
			<IconLink name="google" href="https://mail.google.com/" />
			<IconLink name="facebook" href="https://www.facebook.com/" />
			<IconLink name="linkedin" href="https://www.linkedin.com/in/malgorzata-kowacka/" />
		</div>
	);
};
