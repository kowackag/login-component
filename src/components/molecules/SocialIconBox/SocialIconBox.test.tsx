import { expect, it, describe } from "vitest";
import { render, screen } from "@testing-library/react";
import { SocialIconBox } from "./SocialIconBox";

describe("SocialIconBox", () => {
	it("SocialIconBox exists", async () => {
		render(<SocialIconBox />);

		const box = screen.getByTestId("icon-box");
		expect(box).toBeInTheDocument();
	});

	it("has appropriate numbers of children - links ", async () => {
		render(<SocialIconBox />);
		const box = screen.getByTestId("icon-box");
		expect(box.children.length).toBe(3);
		[...box.children].forEach((el) => expect(el).toHaveAttribute("href"));
	});

	it("renders correctly", () => {
		const { container } = render(<SocialIconBox />);
		expect(container).toMatchSnapshot();
	});
});
