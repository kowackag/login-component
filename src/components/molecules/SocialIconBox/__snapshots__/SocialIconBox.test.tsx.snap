// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SocialIconBox > renders correctly 1`] = `
<div>
  <div
    class="flex justify-center"
    data-testid="icon-box"
  >
    <a
      aria-label="google"
      class="mx-2 my-2"
      data-testid="link"
      href="https://mail.google.com/"
      target="_blank"
    >
      <svg
        height="36px"
        width="36px"
      >
        <use
          href="/google.svg#google"
        />
      </svg>
    </a>
    <a
      aria-label="facebook"
      class="mx-2 my-2"
      data-testid="link"
      href="https://www.facebook.com/"
      target="_blank"
    >
      <svg
        height="36px"
        width="36px"
      >
        <use
          href="/facebook.svg#facebook"
        />
      </svg>
    </a>
    <a
      aria-label="linkedin"
      class="mx-2 my-2"
      data-testid="link"
      href="https://www.linkedin.com/in/malgorzata-kowacka/"
      target="_blank"
    >
      <svg
        height="36px"
        width="36px"
      >
        <use
          href="/linkedin.svg#linkedin"
        />
      </svg>
    </a>
  </div>
</div>
`;
